packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -875,7 +878,24 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
         };
       }
     } else {
-      throwBadResponse(stateP, res);
+      const left = res.left;
+      if (isTypeof(left, 'es_response_too_large')) {
+        const batchSize = Math.floor(stateP.batchSize / 2);
```

Comment:
Could we ever have an edge case of 1 document that exceeds max string length driving batchSize down to `0`. Just wondering if we should add a constraint that batchSize can never be <1 (if it is then we need manual intervention).

Reply:
Not sure how we'd end up in a situation like that if our NodeJS is the only writer to our SO indices so might not be worth handling explicitly.